js/test/test.js

```diff
@@ -141,25 +171,46 @@ async function test (methodName, exchange, ... args) {
     }
 }
 
-async function testSymbol (exchange, symbol) {
-
-    await test ('loadMarkets', exchange);
-    await test ('fetchCurrencies', exchange);
-    await test ('fetchTicker', exchange, symbol);
-    await test ('fetchTickers', exchange, symbol);
-    await test ('fetchOHLCV', exchange, symbol);
-    await test ('fetchTrades', exchange, symbol);
-
-    if (exchange.id === 'coinbase') {
-
-        // nothing for now
+async function testSafe(methodName, exchange, ...args) {
+    try {
+        await testMethod(methodName, exchange, ...args);
+        return true;
+    } catch (e) {
+        return false;
+    }
+}
 
-    } else {
 
-        await test ('fetchOrderBook', exchange, symbol);
-        await test ('fetchL2OrderBook', exchange, symbol);
-        await test ('fetchOrderBooks', exchange);
+async function testSymbol (exchange, symbol) {
+    const tests = {
+        'loadMarkets': [exchange],
+        'fetchCurrencies': [exchange],
+        'fetchTicker': [exchange, symbol],
+        'fetchTickers': [exchange, symbol],
+        'fetchOHLCV': [exchange, symbol],
+        'fetchTrades': [exchange, symbol],
+        'fetchOrderBook': [exchange, symbol],
+        'fetchL2OrderBook': [exchange, symbol],
+        'fetchOrderBooks': [exchange],
+        'fetchBidsAsks': [exchange],
+        'fetchFundingRates': [exchange],
+        'fetchFundingRate': [exchange, symbol],
+        'fetchFundingRateHistory': [exchange, symbol],
+        'fetchIndexOHLCV': [exchange, symbol],
+        'fetchMarkOHLCV': [exchange, symbol],
+        'fetchPremiumIndexOHLCV': [exchange, symbol],
+        'fetchStatus': [exchange],
+        'fetchTime': [exchange],
+    };
```

Comment:
I think we might redesign the structure a bit, because this can't withstand the `swapsymbol` here with same key, i.e.
```
'fetchX' : [exchange, spotSymbol],
'fetchX' : [exchange, swapSymbol],
```
because of duplicate keys.

maybe we can redesign into:

```
'fetchX' : [ [exchange, spotSymbol],  [exchange, swapSymbol], ]
```

Reply:
@ttodua yeah I agree but to be a bit more explicit why not:
```Javascript
'fetchX' : {
    'exchange':exchange,
     'symbols': [spotSymbol, swapSymbol]
}
```

Reply:
I still suspect that because of variety of arguments that might be needed, we might still have some kind of separation. for example, in some cases, spot market might not need extra parameter, while contracts market might need additional parameter (like `marginMode: cross/isolated` as it's required in huobi). but yeah, such cases might be rare and ok, let's go with symbols-array.